libcloud/compute/drivers/vsphere.py

```diff
@@ -12,543 +12,216 @@
-        new_node = self._to_node(vm=new_vm)
-
-        return new_node
-
-    @wrap_non_libcloud_exceptions
-    def ex_migrate_node(self, node, resource_pool=None, host=None,
-                        priority='default'):
-        """
-        Migrate provided node to a new host or resource pool.
-
-        :param node: Node to clone.
-        :type node: :class:`libcloud.compute.base.Node`
-
-        :param resource_pool: ID of the target resource pool to migrate the
-                              node into.
-        :type resource_pool: ``str``
-
-        :param host: Target host to migrate the host to.
-        :type host: ``str``
-
-        :param priority: Migration task priority. Possible values: default,
-                         high, low.
-        :type priority: ``str``
-
-        :return: True on success.
-        :rtype: ``bool``
-        """
-        vm = self._get_vm_for_node(node=node)
-        vm.migrate(priority=priority, resource_pool=resource_pool, host=host)
-
-        return True
+    def __init__(self, host, username, password, verify_ssl=True):
+        context = ssl.create_default_context()
+        if not verify_ssl:
+            context.check_hostname = False
+            context.verify_mode = ssl.CERT_NONE
 
-    @wrap_non_libcloud_exceptions
-    def reboot_node(self, node):
-        vm = self._get_vm_for_node(node=node)
-        vm.reset()
-
-        return True
-
-    @wrap_non_libcloud_exceptions
-    def destroy_node(self, node, ex_remove_files=True):
-        """
-        :param ex_remove_files: Remove all the files from the datastore.
-        :type ex_remove_files: ``bool``
-        """
-        ex_remove_files = False
-        vm = self._get_vm_for_node(node=node)
-
-        server = self.connection.client
-
-        # Based on code from
-        # https://pypi.python.org/pypi/pyxenter
-        if ex_remove_files:
-            request = VI.Destroy_TaskRequestMsg()
-
-            _this = request.new__this(vm._mor)
-            _this.set_attribute_type(vm._mor.get_attribute_type())
-            request.set_element__this(_this)
-            ret = server._proxy.Destroy_Task(request)._returnval
-            task = VITask(ret, server)
-
-            # Wait for the task to finish
-            status = task.wait_for_state([task.STATE_SUCCESS,
-                                          task.STATE_ERROR])
-
-            if status == task.STATE_ERROR:
-                raise LibcloudError('Error destroying node: %s' %
-                                    (task.get_error_message()))
-        else:
-            request = VI.UnregisterVMRequestMsg()
+        try:
+            self.conn = connect.SmartConnect(host=host,
+                                             user=username,
+                                             pwd=password,
+                                             sslContext=context)
+            atexit.register(connect.Disconnect, self.conn)
```

Comment:
Is there a particular justification for using atexit? Probably should be commented here if there is. The connection should be already getting closed when gc cleans this class up.

What the atexit handler in practice is going to end up doing is keep a reference on self.conn, keeping it from being cleaned up normally and thus then deferring it to program shutdown - effectively leaking sockets.